Restructure Powertrain Network Viewer for Upcoming GUI Enhancements (#112)
florentLutz/FAST-OAD-CS23-HE: docs/documentation/powertrain_builder/pt_visual/sort_mechanism.rst

.. _sort_mechanism:

=================
Hierarchy sorting
=================

> The sorting mechanism inspired by the `Tutte's drawing algorithm <https://www.youtube.com/watch?v=mEzPPMhR8XE&t=45s>`_

Reviewer: For future PR where we change the documentation, I suggest we manually compile a version of the read-the-docs targetting the branch and add it to the PR description so that we can have a preview of what it'll look like. This would lead to this in this case: https://fast-oad-cs23-he.readthedocs.io/en/dev_pyvis_to_holoviews/

> is utilized before each visualization in a `static html <https://www.geeksforgeeks.org/computer-networks/difference-between-static-and-dynamic-web-pages/>`_

Reviewer: You don't have to specify about the static html imo you can jsut end the sentence with `utilized before each visualization.`

format. This prevents connection crossing between two layers and allows a more organized presentation of the powertrain
architecture.

Tutte's drawing algorithm
=========================
The Tutte spring theorem, proved by W. T. Tutte :cite:`tutte:1963` in 1963, provides an algorithmic approach with unique

> solution for drawing 3-connected planar graphs. The following garphe is an example of a 3-connected planar graphs.

Reviewer: ```suggestion
solution for drawing 3-connected planar graphs. The following graph is an example of a 3-connected planar graphs.
```
Is the example really worth keeping ?

.. image:: ../../../img/tuttes.svg
    :width: 600
    :align: center

This method has since been widely applied in computing embeddings of complex planar polygons when the boundary nodes are
appropriately selected.

.. _ref-level-sort:

Reference level
===============
The reference level definition is one of the two preparation steps before the actual sorting. Two approaches are
considered to define the root of the powertrain architecture tree, from-storage and from-propulsor. The from-storage

> approach (default) sets all the energy storage components into the base reference level of the whole architecture. The

Reviewer: ```suggestion
approach (default) sets all the energy storage components as the reference level of the whole architecture. The
```

highlighted layer represented the base of the hierarchy.

.. image:: ../../../img/from_storage.svg
    :width: 600
    :align: center

Similar as the from-storage approach, the from-propulsor approach sets the propulsor component as the top reference
level for the whole architecture.

.. image:: ../../../img/from_propulsor.svg
    :width: 600
    :align: center

Boundary nodes
==============
The boundary nodes definition is a critical step for the boundaries of the Tutte's drawing that ensures non-trivial

> solution from the linear system. The boundary nodes are consisted with the nodes in the reference layer and the farthest

Reviewer: ```suggestion
solution from the linear system. The boundary nodes consist of the nodes in the reference layer and the farthest
```
